onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc

```diff
@@ -1335,9 +1339,13 @@ Status MIGraphXExecutionProvider::Compile(const std::vector<FusedNodeAndGraph>&
 
     if (dump_model_ops_) {
       std::string onnx_name = fused_node.Name() + ".onnx";
-      std::ofstream ofs(onnx_name);
+      std::ofstream ofs(onnx_name, std::ios::binary);
+      if (!ofs.is_open()) {
+        ORT_THROW("Failed to open file to dump ONNX model: " + model_name);
```

Comment:
This is a good fix. For even more robust error handling, you might consider checking the stream's state after closing the file (e.g., `if (!ofs)`). This would catch potential errors during the final flush to disk, which could result in a corrupt or incomplete file. However, I see that not checking the return of `close()` is a common pattern in this codebase, so this is more of a suggestion for hardening the code rather than a required change for this PR.